### ISSUE TITLE
Jpeg orientation is lost

### ISSUE THREAD

**Issue by flbaue** [DavBfr/dart_pdf#178]
I want to add jpeg images to a page and I am using ```PdfImage.jpeg(...)``` to do this. After that I am adding the PdfImage to the page with the ```Image(...)``` widget.

Unfortunatly, the image orientation seems to be lost. And so images end up in the wrong orientation on the page. The images were taken on the same iPhone device in portrait. 

Inside the Pdf they show up in landscape.

**Comment by abaker2010** [comment]
I am also having an issue with the jpeg images. When I am trying to build a page for each image, some of the images are flipped completely upside down. Is there any news on this issue?

**Comment by DavBfr** [comment]
This looks like exif rotation information. This is not used.
You could use this library: https://pub.dev/packages/exif and rotate the image depending on the orientation attribute.

**Comment by abaker2010** [comment]
Okay I will give it a try! Thank you for the response and the suggestion.

**Comment by abaker2010** [comment]
Okay so you were absolutely right about the use of the exif data in the image. I wasn't able to get that specific library to work for my usages but I did get another one to work. Florian give this lib a try it is a really easy one to use. 

Library: https://pub.dev/packages/flutter_image_compress

Example usage for my purpose:

`
    var tmpPath = file.path;
    var newName = "tmp_${(basename(tmpPath)).replaceAll(" ", "")}";
    var newPath = tmpPath.replaceAll(basename(tmpPath), newName);           
`
`    return await FlutterImageCompress.compressAndGetFile(
        tmpPath,
        newPath,
        quality: 100,
    );
`

**Comment by DavBfr** [comment]
Very good!
Do you know if Flutter is doing the rotation automatically?
If yes, I could add the same thing on this library. Otherwise, we can close this issue.

**Comment by abaker2010** [comment]
Thanks! I believe they are rotation automatically due to the exif data. I saw mentions for platform specific orientation in the data. But the plugin mentioned above does return a striped exif copy of the image to be used, with the option for new orientation of the returned image using the rotate property when building the new image.

**Comment by flbaue** [comment]
@abaker2010 thanks, I am also using that library as a workaround. It uses the andorid/ios native image processing wich considers the exif data and saves the image in a corrected orientation.

@DavBfr Flutter/Dart is doing the orientation correction if you instantiate an image with 
```Dart
final codec = await instantiateImageCodec(bytesFromFile);
final frame = await codec.getNextFrame();
final image = frame.image;
```
But the issue here is, that this seems to take a lot of resources (memory and cpu), so for a single photo or smaller images it's fine. But when you try to do it with multiple full size photos, like I did, it will get very slow or you might get meory limit exceptions from the plattform.

**Comment by DavBfr** [comment]
Thanks,
I'll keep this open and try to fix that.

**Comment by flbaue** [comment]
@DavBfr Thanks for your effort! Maybe it helps to not instantiate an image directly but to use the ```MemoryImage``` class. It's basically doing the same, but as an ```ImageProvider```it is also using the global image cache.

**Comment by DavBfr** [comment]
I updated the master branch. Could you try by adding this to your `pubspec.yaml`:

```yaml
dependency_overrides:
  pdf:
    git:
      url: https://github.com/DavBfr/dart_pdf
      path: pdf
```

then load your images directly from the camera files.

**Comment by flbaue** [comment]
I just tried it with `PdfImage.jpeg(...)` without providing the orientation myself, and it looks very good! 🎉 Thank you for your work!

**Comment by DavBfr** [comment]
Great!
Don't hesitate to [Buy Me A Coffee](https://www.buymeacoffee.com/JORBmbw9h).

**Comment by DavBfr** [comment]
Implemented in pdf 1.3.25
